libmeepgeom/ring-ll.cpp

```diff
@@ -151,6 +151,9 @@ int main(int argc, char *argv[])
 
   master_printf("all harminv results match reference values\n");
 
+  // this seems to be necessary to prevent failures
+  all_wait();
```

Comment:
Without this line, the results are erratic: sometimes everything works fine, sometimes the master process produces proper output but one of the other processes aborts with an MPI error message, and sometimes the abort happens before the master process ever produces output.

Could there be something non-MPI-friendly in harminv?